### ISSUE TITLE
/paths and $refs of paths - only the first is displayed

### ISSUE THREAD

**Issue by alaricljs** [swagger-api/swagger-ui#3727]
Placing a path ref in paths: prior the path it references will hide that path and only the ref path will be displayed.  Placing the path ref after the path it references will hid the ref path.  Can be reproduced in swagger-editor with the petstore example and this:

    /fakepet:
      $ref: "#/paths/~1pet"

| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  Bug
| Which Swagger/OpenAPI version?  | 2.0 and 3.0.0 tested
| Which Swagger-UI version?       | "3.1.9/gf311404-dirty" in swagger-editor
| How did you install Swagger-UI? | npm i swagger-editor-dist
| Which browser & version?        | Chrome 61.0.3163.100 (Official Build) (64-bit)
| Which operating system?         | Arch Linux (full update Oct-2-2017)

### Expected Behavior
Both paths (and additional refs to the same path) should be fully rendered, or as a change to current UI behavior all the paths that lead to the same operations should render together on the spec header. eg:
```    
    POST /pet                          Add a new pet to the store
         /fakepet
```

### Current Behavior
Only the first path in the file is rendered whether it's a ref or the original path.

### Context
I am trying to transition developers from one URI spec to another and it's important that they be able to reference the docs for both specs until the old one is killed off.  They and management love the GUI look rather than reading actual specs.

**Comment by shockey** [comment]
Yeah, I'm seeing this too. I think this has something to do with `operationId` reuse.

Consider this example:

```yaml
paths:
  /pet:
    post:
      tags:
      - "pet"
      summary: "Add a new pet to the store"
      description: ""
      operationId: "addPet"
  /fakepet:
    $ref: "#/paths/~1pet"
```

This would yield the following after `$ref` resolution:

```yaml
paths:
  /pet:
    post:
      tags:
      - "pet"
      summary: "Add a new pet to the store"
      description: ""
      operationId: "addPet"
  /fakepet:
    post:
      tags:
      - "pet"
      summary: "Add a new pet to the store"
      description: ""
      operationId: "addPet"
```

As you can see, both POST operations end up with the `addPet` operationId. This is invalid, per the Swagger 2.0 specification:

> `operationId`: Unique string used to identify the operation. The id MUST be unique among all operations described in the API.

---

Removing the `operationId` causes Swagger-UI to render both operations:

<img width="1432" alt="messages image 1723745046" src="https://user-images.githubusercontent.com/680248/31152376-9e6fede0-a850-11e7-8a12-12b5f523a440.png">

That would solve your problem, but would require you to ditch your `operationId`s, which isn't great.

As I said, your $ref creates an invalid spec... but I think your end goal is a fair one.

@webron, can you suggest a way to sidestep this `operationId` problem within the definition?

Beyond that, we could modify Swagger-UI to track operations by `path + method` instead of `operationId`.
